src/theory/builtin/generic_op.cpp

```diff
@@ -0,0 +1,320 @@
+    {
+      const FloatingPointToFPSignedBitVector& fsbv =
+          n.getConst<FloatingPointToFPSignedBitVector>();
+      indices.push_back(nm->mkConstInt(fsbv.getSize().exponentWidth()));
+      indices.push_back(nm->mkConstInt(fsbv.getSize().significandWidth()));
+    }
+    break;
+    case FLOATINGPOINT_TO_FP_FROM_REAL:
+    {
+      const FloatingPointToFPReal& fr = n.getConst<FloatingPointToFPReal>();
+      indices.push_back(nm->mkConstInt(fr.getSize().exponentWidth()));
+      indices.push_back(nm->mkConstInt(fr.getSize().significandWidth()));
+    }
+    break;
+    case FLOATINGPOINT_TO_SBV:
+    {
+      const FloatingPointToSBV& fsbv = n.getConst<FloatingPointToSBV>();
+      indices.push_back(nm->mkConstInt(Rational(fsbv)));
+    }
+    break;
+    case FLOATINGPOINT_TO_UBV:
+    {
+      const FloatingPointToUBV& fubv = n.getConst<FloatingPointToUBV>();
+      indices.push_back(nm->mkConstInt(Rational(fubv)));
+    }
+    break;
+    case FLOATINGPOINT_TO_SBV_TOTAL:
+    {
+      const FloatingPointToSBVTotal& fsbv =
+          n.getConst<FloatingPointToSBVTotal>();
+      indices.push_back(nm->mkConstInt(Rational(fsbv)));
+    }
+    break;
+    case FLOATINGPOINT_TO_UBV_TOTAL:
+    {
+      const FloatingPointToUBVTotal& fubv =
+          n.getConst<FloatingPointToUBVTotal>();
+      indices.push_back(nm->mkConstInt(Rational(fubv)));
+    }
+    break;
+    case APPLY_TESTER:
+    {
+      unsigned index = DType::indexOf(n);
+      const DType& dt = DType::datatypeOf(n);
+      indices.push_back(dt[index].getConstructor());
+    }
+    break;
+    case APPLY_UPDATER:
+    {
+      unsigned index = DType::indexOf(n);
+      const DType& dt = DType::datatypeOf(n);
+      unsigned cindex = DType::cindexOf(n);
+      indices.push_back(dt[cindex][index].getSelector());
+    }
+    break;
+    default:
+      Unhandled() << "GenericOp::getOperatorIndices: unhandled kind " << k;
+      break;
+  }
+  return indices;
+}
+
+/** Converts a list of constant integer terms to a list of unsigned integers */
+bool convertToNumeralList(const std::vector<Node>& indices,
+                          std::vector<uint32_t>& numerals)
+{
+  for (const Node& i : indices)
+  {
+    Assert(i.getKind() == CONST_INTEGER);
+    const Integer& ii = i.getConst<Rational>().getNumerator();
+    if (!ii.fitsUnsignedInt())
+    {
+      return false;
+    }
+    numerals.push_back(ii.toUnsignedInt());
+  }
+  return true;
+}
+
+Node GenericOp::getOperatorForIndices(Kind k, const std::vector<Node>& indices)
+{
+  // all indices should be constant!
+  Assert(isIndexedOperatorKind(k));
+  NodeManager* nm = NodeManager::currentNM();
+  if (isNumeralIndexedOperatorKind(k))
+  {
+    std::vector<uint32_t> numerals;
+    if (!convertToNumeralList(indices, numerals))
+    {
+      // failed to convert due to overflow on index
+      return Node::null();
+    }
+    switch (k)
+    {
+      case REGEXP_LOOP:
+        Assert(numerals.size() == 2);
```

Comment:
We should just have a map somewhere that maps index kind to number of indices. This would also make the rewriter more robust.

Reply:
Also, this might be overkill, see my comment about term arguments.